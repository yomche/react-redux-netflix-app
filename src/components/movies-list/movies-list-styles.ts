import styled from 'styled-components';

export const StyledMoviesList = styled.main`
    width: 1000px;
    margin: auto;
    padding-top: 30px;
    padding-bottom: 50px;
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    row-gap: 60px;
    column-gap: 100px;
`;
