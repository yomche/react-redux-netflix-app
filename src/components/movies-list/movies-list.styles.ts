import styled from 'styled-components';

export const StyledContainer = styled.section`
  width: 1200px;
  margin: auto;
`;

export const StyledMoviesList = styled.ul`
  width: 1200px;
  margin: auto;
  padding-left: 0;
  padding-top: 30px;
  padding-bottom: 50px;
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  row-gap: 60px;
  column-gap: 100px;
`;
