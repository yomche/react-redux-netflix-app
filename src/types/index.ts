export type DataType = {
  data: [];
};

export type MoviesDataType = {
  poster_path: string;
  title: string;
  release_date: string;
  genres: string;
  overview: string;
  id: number;
};

export type CurrentMovieType = {
  poster_path: string;
  title: string;
  releaseDate: string;
  overview: string;
  id: number;
};

export type FormStateType = {
  inputValue: string;
  searchType: string;
};
