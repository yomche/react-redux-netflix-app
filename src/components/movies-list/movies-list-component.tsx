import React, { FC } from 'react';
import { StyledMoviesList } from './movies-list-styles';
import { MovieCard } from '../movie-card/movie-card-component';
import { movies } from '../../movies-list';

export const MoviesList: FC = () => (
  <StyledMoviesList>
    {movies.map((movie) => (
      <MovieCard
        poster={movie.poster_path}
        title={movie.title}
        genre={movie.genres[0]}
        date={movie.release_date}
        key={movie.id}
      />
    ))}
  </StyledMoviesList>
);
