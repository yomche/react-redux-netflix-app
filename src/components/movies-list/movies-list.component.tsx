import React, { FC } from 'react';
import { StyledMoviesList, StyledContainer, StyledMoviesNotFound } from './movies-list.styles';
import { MovieCard } from '../movie-card/movie-card.component';

type MoviesListProps = {
  moviesData: [];
  onSetMovieFullInfoType: (currentMovie: {
    posterPath: string;
    title: string;
    releaseDate: string;
    overview: string;
    genre: string;
    id: number;
  }) => React.MouseEventHandler<HTMLLIElement>;
};

export const MoviesList: FC<MoviesListProps> = ({ onSetMovieFullInfoType, moviesData }) => (
  <StyledContainer>
    <StyledMoviesList>
      {moviesData !== undefined &&
        moviesData.map(({ poster_path, title, genres, release_date, overview, id }) => (
          <MovieCard
            onClickToChange={onSetMovieFullInfoType}
            posterPath={poster_path}
            title={title}
            genre={genres[0]}
            releaseDate={release_date}
            overview={overview}
            id={id}
          />
        ))}
      {moviesData.length === 0 && (
        <StyledMoviesNotFound>Sorry, no movies were found</StyledMoviesNotFound>
      )}
    </StyledMoviesList>
  </StyledContainer>
);
