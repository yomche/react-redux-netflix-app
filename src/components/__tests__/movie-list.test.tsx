import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mount } from 'enzyme';

import { MoviesList } from '../movies-list/movies-list.component';
import { MoviesDataType } from '../../types';

describe('< MovieList /> component', () => {
  type MoviesListProps = {
    moviesData: MoviesDataType[];
    setMovieFullInfoType: (currentMovie: {
      poster_path: string;
      title: string;
      release_date: string;
      overview: string;
      genre: string;
      id: number;
    }) => React.MouseEventHandler<HTMLLIElement>;
  };

  const MoviesListMock: MoviesListProps = {
    moviesData: [
      {
        poster_path:
          'https://sun9-72.userapi.com/impf/a-V1H8zLHXqzMb3UY8yzgicFlOVZkHw4TNXeLQ/PHLkRY64ehg.jpg?size=1459x2160&quality=96&sign=1063b2565af2a89c8ae442b61427f9e7&type=album',
        title: 'Black Mirror',
        release_date: '2011-02-07',
        genres: ['Drama', 'Thriller'],
        overview:
          'Set in a world only minutes from our own, Black Mirror unveils how modern technologies can backfire and be used against their makers, every episode set in a slightly different reality with different characters combating different types of technologies.',
        id: 1000,
      },
      {
        poster_path:
          'https://sun9-51.userapi.com/impf/wTwDr3Eh4cZlJPKHiHp90nPQlTlkGfPsAfqFaQ/7VcL1Zco3C4.jpg?size=1458x2160&quality=96&sign=4b7d9e8ab01d00a262d358f8a2e82534&type=album',
        title: 'Chilling Adventures of Sabrina',
        release_date: '2018-12-13',
        genres: ['Fantasy', 'Drama'],
        overview:
          "Reimagines the origin and adventures of Sabrina: the Teenage Witch as a dark coming-of-age story that traffics in horror, the occult and, of course, witchcraft. Tonally in the vein of Rosemary's Baby and The Exorcist, this adaptation finds Sabrina Spellman wrestling to reconcile her dual nature - half-witch, half-mortal - while standing against the evil forces that threaten her, her family and the daylight world humans inhabit.",
        id: 1001,
      },
      {
        poster_path:
          'https://sun9-42.userapi.com/impf/P71f5UItZ2ekG0PzCp16UgWLms_GxmWBzDjREQ/nEkurXb2xWE.jpg?size=1458x2160&quality=96&sign=484ff49b495837b36cf31564231e99ba&type=album',
        title: 'The Witcher',
        release_date: '2019-02-13',
        genres: ['Adventure', 'Fantasy'],
        overview:
          'The Witcher is an American fantasy drama web television series created by Lauren Schmidt Hissrich for Netflix. It is based on the book series of the same name by Polish writer Andrzej Sapkowski. The Witcher follows the story of Geralt of Rivia, a solitary monster hunter, who struggles to find his place in a world where people often prove more wicked than monsters and beasts. But when destiny hurtles him toward a powerful sorceress, and a young princess with a special gift, the three must learn to navigate independently the increasingly volatile Continent.',
        id: 1004,
      },
    ],
    setMovieFullInfoType: jest.fn(),
  };

  it('renders with movies data', () => {
    const { asFragment } = render(
      <Router>
        <MoviesList
          onSetMovieFullInfoType={MoviesListMock.setMovieFullInfoType}
          moviesData={MoviesListMock.moviesData}
        />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders 3 films', () => {
    const component = mount(
      <Router>
        <MoviesList
          onSetMovieFullInfoType={MoviesListMock.setMovieFullInfoType}
          moviesData={MoviesListMock.moviesData}
        />
      </Router>
    );

    expect(component.find('li')).toHaveLength(3);
  });

  it('return "No movies were found" line if there is no movies data', () => {
    const component = mount(
      <MoviesList onSetMovieFullInfoType={MoviesListMock.setMovieFullInfoType} moviesData={[]} />
    );

    expect(component.text()).toBe('Sorry, no movies were found');
  });
});
