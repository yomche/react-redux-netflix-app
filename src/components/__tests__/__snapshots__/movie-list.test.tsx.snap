// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< MovieList /> component renders with movies data 1`] = `
<DocumentFragment>
  <section
    class="StyledContainer-sc-825f46-1 izpdVG"
  >
    <ul
      class="StyledMoviesList-sc-825f46-2 ewWvTM"
    >
      <a
        class="StyledLink-sc-825f46-0 hUVJTR"
        href="/movies/1000"
      >
        <li
          class="StyledMovieCard-sc-1it6lj2-0 iqQZfi"
        >
          <img
            class="MovieCardPoster-sc-1it6lj2-1 kABfhE"
            src="https://sun9-72.userapi.com/impf/a-V1H8zLHXqzMb3UY8yzgicFlOVZkHw4TNXeLQ/PHLkRY64ehg.jpg?size=1459x2160&quality=96&sign=1063b2565af2a89c8ae442b61427f9e7&type=album"
          />
          <h3
            class="StyledMovieCardTitle-sc-1it6lj2-2 jAURuc"
          >
             Black Mirror 
          </h3>
          <div
            class="StyledMovieCardInfo-sc-1it6lj2-5 hGVonm"
          >
            <p
              class="StyledMovieCardGenre-sc-1it6lj2-3 hBpVXW"
            >
               Drama 
            </p>
            <p
              class="StyledMovieCardDate-sc-1it6lj2-4 etwnAe"
            >
               2011-02-07 
            </p>
          </div>
        </li>
      </a>
      <a
        class="StyledLink-sc-825f46-0 hUVJTR"
        href="/movies/1001"
      >
        <li
          class="StyledMovieCard-sc-1it6lj2-0 iqQZfi"
        >
          <img
            class="MovieCardPoster-sc-1it6lj2-1 kABfhE"
            src="https://sun9-51.userapi.com/impf/wTwDr3Eh4cZlJPKHiHp90nPQlTlkGfPsAfqFaQ/7VcL1Zco3C4.jpg?size=1458x2160&quality=96&sign=4b7d9e8ab01d00a262d358f8a2e82534&type=album"
          />
          <h3
            class="StyledMovieCardTitle-sc-1it6lj2-2 jAURuc"
          >
             Chilling Adventures of Sabrina 
          </h3>
          <div
            class="StyledMovieCardInfo-sc-1it6lj2-5 hGVonm"
          >
            <p
              class="StyledMovieCardGenre-sc-1it6lj2-3 hBpVXW"
            >
               Fantasy 
            </p>
            <p
              class="StyledMovieCardDate-sc-1it6lj2-4 etwnAe"
            >
               2018-12-13 
            </p>
          </div>
        </li>
      </a>
      <a
        class="StyledLink-sc-825f46-0 hUVJTR"
        href="/movies/1004"
      >
        <li
          class="StyledMovieCard-sc-1it6lj2-0 iqQZfi"
        >
          <img
            class="MovieCardPoster-sc-1it6lj2-1 kABfhE"
            src="https://sun9-42.userapi.com/impf/P71f5UItZ2ekG0PzCp16UgWLms_GxmWBzDjREQ/nEkurXb2xWE.jpg?size=1458x2160&quality=96&sign=484ff49b495837b36cf31564231e99ba&type=album"
          />
          <h3
            class="StyledMovieCardTitle-sc-1it6lj2-2 jAURuc"
          >
             The Witcher 
          </h3>
          <div
            class="StyledMovieCardInfo-sc-1it6lj2-5 hGVonm"
          >
            <p
              class="StyledMovieCardGenre-sc-1it6lj2-3 hBpVXW"
            >
               Adventure 
            </p>
            <p
              class="StyledMovieCardDate-sc-1it6lj2-4 etwnAe"
            >
               2019-02-13 
            </p>
          </div>
        </li>
      </a>
    </ul>
  </section>
</DocumentFragment>
`;
